### PR TITLE
feat(ramp): add initial sell navigation

### PR DESCRIPTION
**Pull request by wachunei**
## **Description**

This PR refactors and adapts the navigation and hooks from the Ramp flow to support a new type of ramp: off-ramp. This is done by setting a `rampType` property in the current flow. This property will also be used in the hooks that get the data from the ramp SDK to decide wether call an on-ramp method or off-ramp method.

This PR will allow the user to open the sell flow, however **the UI elements will not reflect this**, the entities loaded (supported regions, payment methods, crypto currencies, fiat currencies, etc) will be the corresponding items to the sell flow.

## **Manual testing steps**

1. Go through the buy flow as usual by clicking on Buy buttons. The flow should not change from current production version.
2. As seen on the video, click in the action button in the middle of the bottom section
3. Click on Sell to open the Sell flow
4. Follow the flow up to the redirection from the quote to the provider's website

### **Before**

_n/a_

### **After**

https://github.com/MetaMask/metamask-mobile/assets/1024246/c2880178-5e82-4646-b77a-1c318a3b3819

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by bkirb** [comment]
@wachunei LGTM I verified the buy and sell flows ✅

File: app/components/UI/Ramp/common/components/Quote/Quote.tsx [hunk 1]
```diff
@@ -145,20 +151,45 @@ const Quote: React.FC<Props> = ({
...
+                      toTokenMinimalUnit(
+                        amountOut,
+                        crypto?.decimals ?? 0,
+                      ).toString(),
+                      crypto?.decimals ?? 0,
+                    )}{' '}
+                    {crypto?.symbol}
+                  </>
+                ) : (
+                  `≈ ${renderFiat(amountOut, fiatCode, fiat?.decimals)}`
```
Comment (georgeweiler) [inline app/components/UI/Ramp/common/components/Quote/Quote.tsx:166]: I think we may want `amountOutInFiat ?? price` here

Comment (wachunei) [reply app/components/UI/Ramp/common/components/Quote/Quote.tsx:166]: For sell orders we don't rely on the price, the price as fallback is set for buy orders fiat estimation

**Review by georgeweiler — APPROVED** [review]

**Review by NicolasMassart — COMMENTED** [review]
Not sure if the TODOs are solved in other PRs of this batch, but otherwise, aside from that, LGTM.

File: app/components/UI/Ramp/buy/Views/Checkout.tsx [hunk 2]
```diff
@@ -103,7 +115,11 @@ const CheckoutWebView = () => {
           return;
         }
         const orders = await SDK.orders();
-        const order = await orders.getOrderFromCallback(
+        const getOrderFromCallbackMethod =
+          rampType === RampType.BUY
+            ? 'getOrderFromCallback'
```
Comment (tommasini) [inline app/components/UI/Ramp/buy/Views/Checkout.tsx:120]: nit for future small refactor: `getBuyOrderfromCallback` instead just to be consistent on the order type

File: app/components/UI/Ramp/common/components/RegionModal.tsx [hunk 3]
```diff
@@ -135,7 +137,11 @@ const RegionModal: React.FC<Props> = ({
         setSearchString('');
         return;
       }
-      if (region.unsupported) {
+      if (
```
Comment (tommasini) [inline app/components/UI/Ramp/common/components/RegionModal.tsx:140]: NIT for future refactoring: Could be converted to a function just for naming readable convention

**Review by tommasini — COMMENTED** [review]
Awesome work! 

Just left some comments for future improvements, I will resume here:
* Refactor for differentiate Buy vs Sell variable names around multiple components (usually only the sell logic have the sell on it's name)
* Probably would be cool move the Quotes component to the common fould since it's not buy specific anymore

File: app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx [hunk 4]
```diff
@@ -107,10 +110,18 @@ function Quotes() {
     query: fetchQuotes,
   } = useQuotes(params.amount);
 
-  const filteredQuotes = useMemo(
-    () => quotes?.filter((quote): quote is QuoteResponse => !quote.error) ?? [],
-    [quotes],
-  );
+  const filteredQuotes = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:113]: Definitely, we anticipated having different folders but ended re-using the same views, we will address this later.

File: app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx [hunk 5]
```diff
@@ -306,9 +324,10 @@ function Quotes() {
 
   useEffect(() => {
     if (quotes && !isFetchingQuotes && pollingCyclesLeft >= 0) {
-      const quotesWithoutError = quotes.filter(
-        (quote): quote is QuoteResponse => !quote.error,
-      );
+      const quotesWithoutError = filteredQuotes as (
+        | QuoteResponse
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/buy/Views/Quotes/Quotes.tsx:328]: This is defined in the SDK and is not part of the mobile project scope https://socket.dev/npm/package/@consensys/on-ramp-sdk/files/1.25.2/dist/IOnRampSdk.d.ts#L42 

The `SellQuoteResponse` is a QuoteResponse too, we want to generalize the type here as a both Buy and Sell.

File: app/components/UI/Ramp/buy/hooks/useCryptoCurrencies.ts [hunk 6]
```diff
@@ -21,7 +23,9 @@ export default function useCryptoCurrencies() {
     },
     queryGetCryptoCurrencies,
   ] = useSDKMethod(
-    'getCryptoCurrencies',
+    rampType === RampType.BUY
+      ? 'getCryptoCurrencies'
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/buy/hooks/useCryptoCurrencies.ts:27]: Same explanation as here https://github.com/MetaMask/metamask-mobile/pull/7495#discussion_r1381953099

**Pull request merged**